TwitchDownloaderCLI/Modes/UpdateHandler.cs

```diff
@@ -0,0 +1,192 @@
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return arch switch
+                {
+                    Architecture.X64 => string.Format(origPackageName, "MacOS-x64"),
+                    Architecture.Arm64 => string.Format(origPackageName, "MacOSArm64"),
+                    _ => string.Empty
+                };
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                // TODO: Change to 'linux-musl-x64' when .NET 8+
+                if (RuntimeInformation.RuntimeIdentifier.Contains("musl"))
+                {
+                    return string.Format(origPackageName, "LinuxAlpine-x64");
+                } 
+                else
+                {
+                    return arch switch
+                    {
+                        Architecture.X64 => string.Format(origPackageName, "Linux-x64"),
+                        Architecture.Arm => string.Format(origPackageName, "LinuxArm"),
+                        Architecture.Arm64 => string.Format(origPackageName, "LinuxArm64"),
+                        _ => string.Empty
+                    };
+                }    
+            }
+
+            return string.Empty;
+
+        }
+
+        private static async Task AutoUpdate(string url)
+        {
+            var currentExePath = Environment.ProcessPath;
+            var oldExePath = currentExePath + ".bak";
+            var updateDir = Path.GetDirectoryName(currentExePath);
+            var archivePath = Path.Combine(updateDir, url.Split("/").Last());
+
+            if (string.IsNullOrEmpty(currentExePath))
+            {
+                Console.Error.WriteLine("Internal error: Current executable path is null or empty!");
+            }
+
+            if (string.IsNullOrEmpty(updateDir))
+            {
+                Console.Error.WriteLine("Internal error: Update directory is null or empty!");
+            }
+
+            Console.WriteLine("Downloading update archive...");
+            Stream s = await _client.GetStreamAsync(url);
+
+            // Create downloaded archive file from stream data
+            using (var fs = new FileStream(archivePath, FileMode.OpenOrCreate))
+            {
+                s.CopyTo(fs);
+            }
+
+            // Check for a previous update
+            if (File.Exists(oldExePath))
+            {
+                File.Delete(oldExePath);
+            }
+
+            // Rename current exe
+            File.Move(currentExePath, oldExePath);
+
+            if (File.Exists(Path.Combine(updateDir, "COPYRIGHT.txt")))
+            {
+                File.Delete(Path.Combine(updateDir, "COPYRIGHT.txt"));
+            }
+
+            if (File.Exists(Path.Combine(updateDir, "THIRD-PARTY-LICENSES.txt")))
+            {
+                File.Delete(Path.Combine(updateDir, "THIRD-PARTY-LICENSES.txt"));
+            }
+
+            ZipFile.ExtractToDirectory(archivePath, updateDir);
```

Comment:
The program crashes silently when calling to `ZipFile.ExtractToDirectory`. We might need to include another application specifically for replacing the old executable or use a shell script
![image](https://github.com/user-attachments/assets/881e63da-548d-4568-85dd-16ec4a29f745)

Reply:
I found a workaround.